materials/03-remote-hosting.qmd

```diff
@@ -2,4 +2,203 @@
+
+2.  Click on the `+` icon in the top right corner of your personal dashboard, then `New Repository`
+
+3.  Name your repository “planets”. Further down the page are other options available to us that can be toggled to suit the repository's purpose. For this workshop it's fine to create a "Public" repository however since this repository will be connected to a local repository, it needs to be empty. Therefore, leave “Add a README file” unchecked, and select “None” as options for both “Add .gitignore” and “Add a license.”
+
+4.  Click "Create Repository". As soon as the repository is created, GitHub displays a page with a URL and some information on how to configure your local repository.
+
+![](images/clipboard-3105004005.png)
+
+5.  Make sure HTTPS is selected and then click the clipboard icon to copy the URL. We will use this URL in the next section.
+
+Note: you have effectively done the following on GitHub's servers:
+
+```{bash}
+$ mkdir planets
+$ cd planets
+$ git init
+```
+
+If you remember back to the earlier episode where we added and committed our earlier work on mars.txt, we had a diagram of the local repository which looked like this:
+
+![](../figures/tracking-changes.png)
+
+Now that we have two repositories, we need a diagram like this
+
+![](../materials/images/git-freshly-made-github-repo.svg)
+
+Note that our local repository still contains our earlier work on `mars.txt`, but the remote repository on GitHub appears empty as it doesn’t contain any files yet.
+
+## **3.2 Connecting to (Cloning) a Remote Repository**
+
+Now we connect the two repositories. We do this by making the GitHub repository a [remote](https://swcarpentry.github.io/git-novice/reference.html#remote) for the local repository. Previously we were shown a page which looked like this:
+
+![](images/clipboard-321009940.png)
+
+At the time, we chose to use HTTPS as our security protocol for connecting the two repositories.
+
+**SAY SOMETHING ABOUT HTTPS vs SSH**
+
+Go into the local `planets` repository, and run this command:
+
+```{bash}
+$ git remote add origin https://github.com/<yourusername>/planets.git
+```
+
+which is equivalent to
+
+```{bash}
+$ git remote add origin <paste copied URL here>
+```
+
+`origin` is a local name used to refer to the remote repository. It could be called anything, but `origin` is a convention that is often used by default in git and GitHub, so it’s helpful to stick with this unless there’s a reason not to.
+
+We can check that the command has worked by running `git remote -v`:
+
+```{bash}
+$ git remote -v
+```
+
+``` bash-output-box
+origin   https://github.com/<yourusername>/planets.git (fetch)
+origin   https://github.com/<youusername>/planets.git (push)
+```
+
+## **3.2 Push Local Changes to a Remote Repository**
+
+Now that authentication is setup, we can return to the remote. This command will push the changes from our local repository to the repository on GitHub:
+
+```{bash}
+$ git push origin main
```

Comment:
I think we probably need to mention when you do/don't need to include the `origin main` part of the push message. (i.e. basically never, I think the way this was set up it's not even needed here).